doc/api/esm.md

```diff
@@ -83,12 +83,25 @@ All CommonJS, JSON, and C++ modules can be used with `import`.
 Modules loaded this way will only be loaded once, even if their query
 or fragment string differs between `import` statements.
 
-When loaded via `import` these modules will provide a single `default` export
-representing the value of `module.exports` at the time they finished evaluating.
+JSON and C++ addon modules will provide a single `default` export representing
+the value of `module.exports` at the time they finish evaluating.
+
+CommonJS modules, when imported, will be handled in one of two ways. By default
+they will provide a single `default` export representing the value of
+`module.exports` at the time they finish evaluating.
+CJS modules may also provide a boolean `@@esModuleInterop` or `__esModule`
+export indicating that the enumerable keys of `module.exports` should be used
+as named exports.
+In both cases, this should be thought of  like a "snapshot" of the exports at
```

Comment:
Is this a snapshot of the *values*, or just a snapshot of the names?

The latter is necessary, but the former may not be.

Reply:
I thought just saying `exports` was ok since its both the names and the values

Reply:
Right - I’m saying that there’s no need for the values to be snapshotted; just the names.

Reply:
technically it isn't a snapshot, it's just a useful term to describe how it becomes static when assigned in reflection with es imports.

Reply:
It’s pretty important to be precise here :-) i think “a snapshot of the names of the exports”, and indicating that if the values are updated, the resulting imports will update as well (a requirement for APM-like use cases, i understand)